RecoLocalTracker/SiPixelDigiReProducers/plugins/SiPixelDigiMorphing.cc

```diff
@@ -0,0 +1,215 @@
+#ifndef RecoLocalTracker_SiPixelDigiMorphing_SiPixelDigiMorphing_h
+#define RecoLocalTracker_SiPixelDigiMorphing_SiPixelDigiMorphing_h
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/SiPixelDigi/interface/PixelDigi.h"
+
+#include <bitset>
+
+class SiPixelDigiMorphing : public edm::stream::EDProducer<> {
+public:
+  explicit SiPixelDigiMorphing(const edm::ParameterSet& conf);
+  ~SiPixelDigiMorphing() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  void produce(edm::Event& e, const edm::EventSetup& c) override;
+
+private:
+  edm::EDGetTokenT<edm::DetSetVector<PixelDigi>> tPixelDigi;
+  edm::EDPutTokenT<edm::DetSetVector<PixelDigi>> tPutPixelDigi;
+
+  int32_t nrows_;
+  int32_t ncols_;
+  int32_t nrocs_;  // in Phase 1, this is ROCs, but could be any subset of a pixel row
+  int32_t ncols_r_;
+
+  int32_t iters_;
+  int32_t ksize_;
```

Comment:
```suggestion
  const int32_t nrows_;
  const int32_t ncols_;
  const int32_t nrocs_;  // in Phase 1, this is ROCs, but could be any subset of a pixel row
  const int32_t ncols_r_;

  const int32_t iters_;
  const int32_t ksize_;
```
it looks like all of these can be const